1012/bn.md

```diff
@@ -0,0 +1,133 @@
+# ক্যাটাগরি : ডি এফ এস 
+
+### প্রশ্নে কি জিজ্ঞেস করেছে ? 
+
+কাহিনীর অর্থে : একজন রাজা তার ছেলেকে রাজ্য ছাড়তে বলেন । পুত্র উনার কথা মান্য করে রাজ্য ছাড়ে । পরবর্তীতে রাজকুমার এমন একটি জায়গায় যান যেখানে প্রতিটি অঞ্চল হয় পানিতে পূর্ণ 
+অথবা জমিন । রাজকুমার জানতে চান উনার জায়গা হতে কতগুলা জমিন এ উনি যেতে পারবেন 
+
+গাণিতিক অর্থে : একটি nxm গ্রিড দেয়া থাকবে । একটি পজিশন দেয়া থাকবে যে জায়গা থেকে শুরু করে কয়টি সেল এ যেতে পারবো ? একটা সেল থেকে অন্য সেল এ যাওয়া যাবে যদি সেল টি তে '.' থাকে
+
+### যা যা শিখা উচিত এইটা করার আগে : 
+১ [ডি এফ এস  ](http://www.shafaetsplanet.com/?p=973 ) (এটি বি এফ এস দিয়ে ও করা যাবে ) 
+
+## সমাধান 
+
+আমরা প্রথমে একটি nxm ক্যারেক্টার অ্যারে তে আমাদের পুরো গ্রিড স্টোর করবো 
+নিচের উদাহরণ এর মত একটি 2D অ্যারে তে আমরা ক্যারেক্টার গুলা নিব  
+
+![alt text](https://user-images.githubusercontent.com/72943111/202870913-02f03555-4222-4eed-a889-b6aee8117c38.png)
+
+ইনপুট টা কে অ্যারে তে স্টোর করার কোড 
+```
+cin>>m>>n;
+        string s;
+        for(int i=0; i<n; i++)
+        {
+            cin>>s;
+            for(int j=0; j<s.size(); j++)
+            {
+                A[i][j]=s[j];
+            }
+        }
+```
+
+যেহেতু আমাদের @ থেকে শুরু করতে হবে সুতরাং আমাদের কোন সেল এ @ আছে টা কিছু একটা তে স্টোর করে রাখলাম । উদাহরণ হিসাবে উপরের চিত্রে (১,১) সেল টা আমাদের যাত্রা শুরুর সেল 
+```
+for(int i=0; i<n; i++)
+        {
+            cin>>s;
+            for(int j=0; j<s.size(); j++)
+            {
+                A[i][j]=s[j];
+                if(A[i][j]=='@')
+                {
+                    x=i;
+                    y=j;
+                }
+            }
+        }
+```
+
+তারপর আমরা যাত্রা শুরুর বিন্দু থেকে ৪ দিকে যাওয়া শুরু করবো যদি সম্ভব হয় 
+```
+        dfs(x,y);
+```
+তারপর আমরা যাত্রা শুরুর বিন্দু থেকে ৪ দিকে যাওয়া শুরু করবো যদি সম্ভব হয় 
+* আমরা এক জায়গা দুইবার ঘুরব না তাই একটি vis অ্যারে রাখব 
+* আমরা নেগাটিভ ইনডেক্স থেকে বেচে থাকবো 
+* আমরা গ্রিড এর বাইরে যেন না যাই তার দিকে খেয়াল রাখব 
+* আমরা এমন কোন সেল এ যাব না যেখানে * নাই 
+* নতুন কোন সেল এ গেলেই আমরা একটা গ্লোবাল চলক এর মান বাড়াব যা আসলে আমদের উত্তর টা 
+```
+void dfs(int x,int y)
+{
+    if(x==n || y==m ||x==-1 || y==-1)
+        return;
+    if(vis[x][y]==1)
+        return ;
+    if(A[x][y]=='#')
+        return ;
+    cnt++;
+    vis[x][y]=1;
+    dfs(x+1,y);
+    dfs(x,y+1);
+    dfs(x-1,y);
+    dfs(x,y-1);
+}
+```
+### কিছু সতর্কতা
+প্রতি কেস এর পর vis এবং cnt ক্লিয়ার করতে হবে 
```

Comment:
Implementation plans.

Reply:
Okay Sir , I am fixing it

Reply:
I tried to fix it Sir. Can you review it and tell me what to do next